lib/cpp/src/thrift/server/TThreadPoolServer.cpp

```diff
@@ -146,9 +73,12 @@ void TThreadPoolServer::serve() {
       shared_ptr<TProcessor> processor = getProcessor(inputProtocol, outputProtocol, client);
 
       // Add to threadmanager pool
-      shared_ptr<TThreadPoolServer::Task> task(
-          new TThreadPoolServer::Task(*this, processor, inputProtocol, outputProtocol, client));
-      threadManager_->add(task, timeout_, taskExpiration_);
+      shared_ptr<TConnectedClient> pClient(
```

Comment:
use make_shared.  It shaves off an allocation.  For bonus terseness... don't even make a local shared_ptr<TConnectedClient>.  Just pass the make_shared result to threadManager_->add.